pandas/tests/api/test_api.py

```diff
@@ -357,6 +357,33 @@ def test_api_extensions(self):
         self.check(api_extensions, self.allowed_api_extensions)
 
 
+class TestExtensions(Base):
+    # top-level classes
+    classes = [
+        "ExtensionDtype",
+        "ExtensionArray",
+        "ExtensionScalarOpsMixin",
+        "NDArrayBackedExtensionArray",
+    ]
+
+    # top-level functions
+    funcs = [
+        "register_extension_dtype",
+        "register_dataframe_accessor",
+        "register_index_accessor",
+        "register_series_accessor",
+        "take",
+    ]
+
+    # misc
+    misc = ["no_default"]
+
+    def test_api(self):
+        checkthese = self.classes + self.funcs + self.misc
```

Comment:
> im not that familiar with this test file. what is being tested here? https://github.com/pandas-dev/pandas/pull/45544#discussion_r1035479236

looks like this was added following this comment: 

> i am shocked we don't have a tests that asserts this api e.g. test_api.py can you add one ?
https://github.com/pandas-dev/pandas/pull/45544#discussion_r790198820

Reply:
moved this to L339 instead